includes/Classifai/Providers/OpenAI/TextToSpeech.php

```diff
@@ -0,0 +1,323 @@
+		$new_settings[ static::ID ]['api_key']       = $api_key_settings[ static::ID ]['api_key'];
+		$new_settings[ static::ID ]['authenticated'] = $api_key_settings[ static::ID ]['authenticated'];
+
+		if ( $this->feature_instance instanceof FeatureTextToSpeech ) {
+			if ( in_array( $new_settings[ static::ID ]['tts_model'], [ 'tts-1', 'tts-1-hd' ], true ) ) {
+				$new_settings[ static::ID ]['tts_model'] = sanitize_text_field( $new_settings[ static::ID ]['tts_model'] );
+			}
+
+			if ( in_array( $new_settings[ static::ID ]['voice'], [ 'alloy', 'echo', 'fable', 'onyx', 'nova', 'shimmer' ], true ) ) {
+				$new_settings[ static::ID ]['voice'] = sanitize_text_field( $new_settings[ static::ID ]['voice'] );
+			}
+
+			if ( in_array( $new_settings[ static::ID ]['format'], [ 'mp3', 'opus', 'aac', 'flac', 'wav', 'pcm' ], true ) ) {
+				$new_settings[ static::ID ]['format'] = sanitize_text_field( $new_settings[ static::ID ]['format'] );
+			}
+
+			$speed = filter_var( $new_settings[ static::ID ]['speed'] ?? 1.0, FILTER_SANITIZE_NUMBER_FLOAT );
+
+			if ( 0.25 <= $speed || 4.00 >= $speed ) {
+				$new_settings[ static::ID ]['speed'] = sanitize_text_field( $new_settings[ static::ID ]['speed'] );
+			}
+		}
+
+		return $new_settings;
+	}
+
+	/**
+	 * Common entry point for all REST endpoints for this provider.
+	 *
+	 * @param int    $post_id       The post ID we're processing.
+	 * @param string $route_to_call The name of the route we're going to be processing.
+	 * @param array  $args          Optional arguments to pass to the route.
+	 * @return array|string|WP_Error
+	 */
+	public function rest_endpoint_callback( $post_id, string $route_to_call = '', array $args = [] ) {
+		if ( ! $post_id || ! get_post( $post_id ) ) {
+			return new WP_Error( 'post_id_required', esc_html__( 'A valid post ID is required.', 'classifai' ) );
+		}
+
+		$route_to_call = strtolower( $route_to_call );
+		$return        = '';
+
+		// Handle all of our routes.
+		switch ( $route_to_call ) {
+			case 'synthesize':
+				$return = $this->synthesize_speech( $post_id, $args );
+				break;
+		}
+
+		return $return;
+	}
+
+	/**
+	 * Synthesizes speech from a post item.
+	 *
+	 * @param int $post_id Post ID.
+	 * @return string|WP_Error
+	 */
+	public function synthesize_speech( int $post_id ) {
+		if ( empty( $post_id ) ) {
+			return new WP_Error(
+				'openai_text_to_speech_post_id_missing',
+				esc_html__( 'Post ID missing.', 'classifai' )
+			);
+		}
+
+		// We skip the user cap check if running under WP-CLI.
+		if ( ! current_user_can( 'edit_post', $post_id ) && ( ! defined( 'WP_CLI' ) || ! WP_CLI ) ) {
+			return new WP_Error(
+				'openai_text_to_speech_user_not_authorized',
+				esc_html__( 'Unauthorized user.', 'classifai' )
+			);
+		}
+
+		$feature             = new FeatureTextToSpeech();
+		$settings            = $feature->get_settings();
+		$post_content        = $feature->normalize_post_content( $post_id );
```

Comment:
Looks like there's a size limit on the content that we can send (4096 characters). Should we add some protections here that checks if we're above that limit and returns early with an error message?

Reply:
Fixed in [45a2b26](https://github.com/10up/classifai/pull/747/commits/45a2b26040470dfaf084770fbbd2598d8ba4396f)